### PR TITLE
Allow specifying tag of repos

### PR DESCRIPTION
**Pull request by Chiiizzzy**
Signed-off-by: Kaiyuan Hu <kaiyuan.hu@zilliz.com>

### CODE REVIEW

File: towhee/hub/file_manager.py [hunk 1]
```diff
@@ -311,13 +311,12 @@ def get_pipeline(self, pipeline: str, tag: str = 'main', install_reqs=True):
...
-
             pipeline_path = self._cache_name(repo, author, tag) / (file_name + '.yaml')
-
             file_path = self._config.default_cache / 'pipelines' / pipeline_path
-            found_existing = False
 
-            for path in self._config.cache_paths:
+            found_existing = bool(file_path.is_file())
+
+            for path in reversed(self._config.cache_paths):
```
Comment (filip-halt) [inline towhee/hub/file_manager.py:319]: Adding a cache path adds it to the front of the list so reverse can be avoided, this change would look through in the wrong order.

File: towhee/hub/file_manager.py [hunk 2]
```diff
@@ -311,13 +311,12 @@ def get_pipeline(self, pipeline: str, tag: str = 'main', install_reqs=True):
...
             repo = pipeline_split[1]
             file_name = repo.replace('-', '_')
-
             pipeline_path = self._cache_name(repo, author, tag) / (file_name + '.yaml')
-
             file_path = self._config.default_cache / 'pipelines' / pipeline_path
-            found_existing = False
 
-            for path in self._config.cache_paths:
+            found_existing = bool(file_path.is_file())
```
Comment (filip-halt) [inline towhee/hub/file_manager.py:317]: bool(file_path.is_file()) as an unnecessary extra file check.

File: towhee/hub/file_manager.py [hunk 3]
```diff
@@ -345,7 +345,7 @@ def get_pipeline(self, pipeline: str, tag: str = 'main', install_reqs=True):
 
         return file_path
 
-    def get_operator(self, operator: str, tag: str = 'main', install_reqs=True):
+    def get_operator(self, operator: str, tag: str = 'main', install_reqs: bool = True):
```
Comment (filip-halt) [inline towhee/hub/file_manager.py:348]: same edits as previous

File: towhee/hub/file_manager.py (hunk 1 above)
Comment (Chiiizzzy) [reply towhee/hub/file_manager.py:317]: I thought we are searching from the default cache path.

File: towhee/hub/file_manager.py (hunk 2 above)
Comment (Chiiizzzy) [reply towhee/hub/file_manager.py]: Fixed

File: towhee/hub/file_manager.py (hunk 1 above)
Comment (filip-halt) [reply towhee/hub/file_manager.py]: for path in reversed(self._config.cache_paths):
Was talking about this part, dont need the reversed. Cache path is searched in opposite order added with the default being the last one. If user adds c1, c2, c3, cache paths list would be [c3, c2, c1] with default being c1. Default is the last checked.

**Comment by junjiejiangjjj** [comment]
If we need access repo, use `RepoManager`, so first, finish the `RepoManger`, and use it instead of the `git` package.

**Pull request merged**